### PR TITLE
Allow an initial deposit address to be given via query string.

### PR DESCRIPTION
**Pull request by miracle2k**
This allows the start the redemption flow with a deposit address already prefilled, for example:

```
http://localhost:3001/redeem/?deposit=0xecfdabdd933799e524e5e21c6672213a1f22e9dd
```

This would enable third party websites & explorer to link to the redemption flow for deposits they know are redeemable.

Pinging @mhluongo as this was his idea.

### CODE REVIEW

File: src/components/redemption/Start.js [hunk 1]
```diff
@@ -21,7 +22,14 @@ const Start = ({ saveAddresses, resetState, openWalletModal }) => {
...
   }
-  const [depositAddress, setDepositAddress] = useState(initialAddressState)
+
+  const location = useLocation()
+  const searchParams = new URLSearchParams(location.search)
+
+  const [depositAddress, setDepositAddress] = useState({
+    ...initialAddressState,
+    address: searchParams.get("deposit") || "",
+  })
```
Comment (Shadowfiend) [inline src/components/redemption/Start.js:32]: Thoughts on routing `/deposit/:address/redeem` to this page? That will match our other routes and set us up for some of the stuff in the new design while maintaining route stability, which is desirable.

Comment (miracle2k) [reply src/components/redemption/Start.js:32]: I like it - I approached it like this initially, wrapped in Loadable, but ended up with an infinite loop. I'll have another go.

**Comment by miracle2k** [comment]
The url is now as you suggested, and I also added an initial validation run to the address.

**Review by Shadowfiend — APPROVED** [review]
Gave it a local test drive, things look good, let's do it!

**Pull request merged**